hooks/useAuthStore.ts

```diff
@@ -49,10 +58,12 @@ export const useAuthStore = create<AuthState>((set) => ({
       set({ isLoading: true, error: null });
       const { data, error } = await supabase.auth.signUp({ email, password });
       if (error) throw error;
-      set({ session: data.session, isLoading: false });
+      set({ isLoading: false, error: null });
+      return 'Please check your email to verify your account.';
     } catch (error) {
       console.error('Error signing up:', error);
       set({ isLoading: false, error: 'Failed to sign up' });
+      throw error;
```

Comment:
Why are we throwing this error here? The reason for throwing an error is to bail out of a method and hit the next catch block but this is doing nothing useful.

Reply:
Probably what we should do here for now is set some variables signUpError and signInError to the two different errors and try to surface them in the code.

Reply:
I think this was a Claude suggestion as some kind of failsafe when I asked it to check my code. I'll be more careful not to take it at it's word!